### PR TITLE
Bug 1159784 - [gaia-header] lint fix and add auto run lint command in travis

### PR DESCRIPTION
**Pull request by gasolin**
* use gaia  .jshintrc
* add dependency badge in README
* fix lints and pass unit tests
* update packages
* add `npm run lint` command for jshint check
* run lint in travis

### CODE REVIEW

File: gaia-header.js [hunk 1]
```diff
@@ -276,24 +279,27 @@ module.exports = component.register('gaia-header', {
...
     // promise, or make a new one
-    return this.unresolved[key] = this.unresolved[key] || new Promise((resolve) => {
-      this.pending[key] = this.nextTick(() => {
-        var styles = this._titleStyles;
-        var els = this.els.titles;
-
-        [].forEach.call(els, (el, i) => {
-          if (!styles[i]) { return debug('exit'); }
-          this.setTitleStyle(el, styles[i]);
+    if (!this.unresolved[key]) {
```
Comment (wilsonpage) [inline gaia-header.js]: I'd prefer:

```js
if (this.unresolved[key]) { return this.unresolved[key]; }
```

Comment (gasolin) [inline gaia-header.js]: sure, modified :)

**Comment by wilsonpage** [comment]
This needs a rebase.

File: gaia-header.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
+/* jshint node:true */
+/* globals define */
```
Comment (wilsonpage) [inline gaia-header.js]: Can these be defined in the `.jshintrc`?

File: test/test.js [hunk 3]
```diff
@@ -132,18 +134,21 @@ suite('GaiaHeader', function() {
     });
 
     test('It doesn\'t throw when there is no header', function() {
-      var el = new GaiaHeader();
+      GaiaHeader();
     });
 
-    test('It shouldn\'t run if there is no title `textContent` (ie. before l10n)', function(done) {
+    test('It shouldn\'t run if there is no title `textContent` ' +
```
Comment (wilsonpage) [inline test/test.js]: IMO this makes test harder to read.

File: test/test.js [hunk 4]
```diff
@@ -1020,7 +1068,8 @@ suite('GaiaHeader', function() {
   });
 
   test('it clears pending async `setTitleStyles`', function(done) {
-    this.dom.innerHTML = '<gaia-header action="menu"><h1>Title</h1></gaia-header>';
+    this.dom.innerHTML = '<gaia-header action="menu"><h1>Title</h1>' +
+      '</gaia-header>';
     var el = this.dom.firstElementChild;
 
     el.runFontFit();
```
Comment (wilsonpage) [inline test/test.js]: Overall I think the line wrapping in tests to conform to 80 char maxlen just makes things harder to read and maintain. I vote to opt out of this particular rule in test files.

File: gaia-header.js (hunk 2 above)
Comment (gasolin) [inline gaia-header.js]: I think it worth to keep `.jslintrc` the same with other `gaia` repo, so there's no surprise between any sub projects

File: test/test.js (hunk 4 above)
Comment (gasolin) [inline test/test.js]: I think it not worth to break the rules cross gaia, but I could set `jshint maxlen:100` for test files if you prefer that.

**Comment by gasolin** [comment]
rebased

File: gaia-header.js (hunk 2 above)
Comment (wilsonpage) [inline gaia-header.js]: IMO Gaia should not be linting outside dependencies :) If the project is not Mozilla owned then the coding convention will likely be different and linting will fail. You wouldn't lint your `node_modules` folder in a Node project.

Comment (wilsonpage) [inline gaia-header.js]: I do agree it's sensible that Gaia related projects share similar coding conventions, but it shouldn't be Gaia's job to enforce these conventions in its own build.

File: .jshintrc [hunk 5]
```diff
@@ -1,6 +1,33 @@
 {
-  "laxbreak": true,
+  "camelcase": false,
+  "curly": true,
+  "forin": false,
+  "latedef": "nofunc",
+  "newcap": false,
+  "noarg": true,
+  "node": true,
```
Comment (gasolin) [inline .jshintrc]: put extra `node: true` here

File: test/test.js [hunk 6]
```diff
@@ -1,11 +1,9 @@
-/*global window,assert,suite,setup,teardown,sinon,test*/
-/*jshint esnext:true*/
-
+/* jshint maxlen:100 */
```
Comment (gasolin) [inline test/test.js]: expand line width to 100 chars

**Comment by gasolin** [comment]
@wilsonpage I've update the PR with maxlen: 100 in test file and put `node: true` in `jshintrc`. Could you kindly review it again?

**Pull request merged**